gcp-deployer/README.md

```diff
@@ -104,3 +104,27 @@ The yaml is unmarshalled into a [machinesetup.configList](../cloud/google/machin
 Right now, the startup scripts must be bash scripts. 
 There is a set of environment variables that is concatenated to the beginning of the startup script.
 You can find them [here](../cloud/google/metadata.go) and use them in your startup script as needed.
+
+## Contributing
+
+### Building
+To build a binary, run the following command from this folder.
+
+```
+go build .
+```
+
+### Testing
+When changing this application, you will often end up modifying other packages above this folder in the project tree. Before 
+opening any PR you should run all the unit tests in the repository. To run the unit tests, run the following command from the 
+root, ```cluster-api```, folder of the repo.
+
+```
+go test ./...
```

Comment:
On a related note, since the integration test is setup like a go test, I strongly suspect the current presubmit will try to run the integration test (and probably fail). I think either the integration test or the presubmit script needs to be adjusted so presubmits work.

Reply:
Especially since it is not likely you have a installed and configured gcloud on the job that is running the presubmit.

Reply:
It won't run the test - I've confirmed on the command line that "go test ./..." does not run this test. The integration tag is required.

I do plan to get this running out of our CI / build, but I'd like to follow up with that.